chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -264,6 +264,23 @@ def self.unclaimed(node, include_mounted=false)
           end
         end
 
+        # return the disks that are unclaimed and prepared for given role (or with no role)
+        # (sort the result so the ones with exact role come first)
```

Comment:
I don't know if that's how it should work. I haven't given this any deep thought, but I would say that it should return the unclaimed disks with the intended role, or if that's empty, then the unclaimed disks with no intended role.

For instance, I probably don't want to have cinder take all cinder disks + all unassigned disks.